File: backend/src/models/expense.ts
import mongoose, { Types, type HydratedDocument } from "mongoose";

const expenseSchema = new mongoose.Schema(
  {
    amount: {
      type: Number,
      required: true,
    },
    title: {
      type: String,
      required: true,
    },
    tripId: {
      type: mongoose.Schema.Types.ObjectId,
      ref: "Trip",
      required: true,
    },
  },
  { timestamps: true },
);

export type ExpenseDocument = HydratedDocument<{
  amount: number;
  title: string;
  trip: Types.ObjectId;
}>;

const Expense = mongoose.model("Expense", expenseSchema);
export default Expense;

File: expense_splitter/lib/core/entities/expense.dart
class Expense {
  final String id;
  final double amount;
  final String title;
  final String tripId;

  Expense({
    required this.id,
    required this.amount,
    required this.title,
    required this.tripId,
  });

  factory Expense.fromJson(Map<String, dynamic> json) {
    return Expense(
      id: json['_id'] ?? '',
      title: json['title'] ?? '',
      amount: json['amount'] ?? '',
      tripId: json['tripId'] ?? '',
    );
  }

  Map<String, dynamic> toJson() {
    return {'_id': id, 'amount': amount, 'title': title, 'tripId': tripId};
  }
}
